tests/jdom/tester.ts

```diff
@@ -0,0 +1,191 @@
+import { JDBus } from "../../src/jdom/bus"
+import JDServiceProvider from "../../src/jdom/serviceprovider"
+import { EVENT, DEVICE_ANNOUNCE } from "../../src/jdom/constants";
+import { mkBus } from "../testutils";
+
+import { JDEvent } from "../../src/jdom/event";
+import { JDDevice } from "../../src/jdom/device";
+import RoleManager from "../../src/servers/rolemanager"
+import JDServiceServer from "../../src/jdom/serviceserver";
+import { assert } from "../../src/jdom/utils";
+import { JDService } from "../../src/jdom/service";
+
+
+// Set equals is not a built-in operation.
+function setEquals<T>(set1: Set<T>, set2: Set<T>): boolean {
+    if (set1.size != set2.size) {
+        return false
+    }
+    set1.forEach(value => {
+        if (!set2.has(value)) {
+            return false
+        }
+    })
+    return true
+}
+
+
+interface ServerDevice {
+    server: JDServiceServer,
```

Comment:
is there ever a case where you'd want to test a device with multiple services?

Reply:
I'm not sure? But I imagine that in the common case testing will focus on services instead of device groupings, and that should be as easy as possible?